lib/worker.py

```diff
@@ -327,7 +329,23 @@ def run_loop(self, task_queue, result_queue):
                 break
 
             result_queue.put(self.current_task(task_id))
-            short_status = self.run_task(task_id)
+            if is_fragile:
+                short_status = ''
+                retries_left = self.suite.fragile_retries()
+                # let's run till short_status became 'pass'
+                while short_status != 'pass' and retries_left >= 0:
+                    # print message only after some fails occurred
+                    if re.match(r'^fail:.*', short_status):
```

Comment:
This commit does not change `short_status` from just `fail` to `fail:<checksum>`, so the regexp looks more as part of the next commit.

Reply:
Right, corrected.